Make locale suggestions notice at login/signup less obtrusive

chriskmnds opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Fixes https://github.com/Automattic/wp-calypso/issues/100426

## Proposed Changes

Make the locale suggestions notice a little less profound on signup/login pages.

## Media

| WPCOM Before | WPCOM After |
|--------|--------|
| <img width="500" alt="Screenshot 2025-03-12 at 6 53 50 PM" src="https://github.com/user-attachments/assets/b36f5809-82b3-455e-be15-5a9e7314ace0" /> | <img width="500" alt="Screenshot 2025-03-12 at 6 53 27 PM" src="https://github.com/user-attachments/assets/06ecff50-b2fb-4a7c-816f-02cfbcd07c1f" /> | 

| Crowdsignal | VIP | Woo |
|--------|--------|--------|
| <img width="400" alt="Screenshot 2025-03-13 at 6 49 46 PM" src="https://github.com/user-attachments/assets/32ac66fc-59bf-48d3-ba6e-f0488f2ffbcb" /> | <img width="400" alt="Screenshot 2025-03-13 at 7 00 40 PM" src="https://github.com/user-attachments/assets/398414bd-3cd4-4385-aee3-2d097a168c2a" /> | <img width="400" alt="Screenshot 2025-03-13 at 6 50 21 PM" src="https://github.com/user-attachments/assets/953a486e-d9fd-43c0-8ff2-59220db4ac4d" /> |

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

Addresses https://github.com/Automattic/wp-calypso/issues/100426

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go to `/log-in/el` and observe the notice shown at the top
* There are several login screens and variations. See comment below for a run down of all of them along with caveats: https://github.com/Automattic/wp-calypso/pull/101254#issuecomment-2718683901
* For instructions and direct local links to access these screens, see https://github.com/Automattic/wp-calypso/pull/100807

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

chriskmnds committed:
Make locale suggestions at login/signup less obtrusive

chriskmnds committed:
icon color

ryelle commented:
For some reason, the globe is green when I test, locally and on the calypso live link, on Firefox and Chrome.

<img width="514" alt="Screenshot 2025-03-12 at 1 30 24 PM" src="https://github.com/user-attachments/assets/5b9cd8a9-34c2-4e77-88e1-c7808edb36cc" />

I've gone through the list of different login screens (from #100807, h/t @adamwoodnz for pulling those together), and there are a few issues. The new compact notice is missing the bottom margin, so it collides with the title on a few of them. It also blends in on Crowdsignal & VIP's screens. This fixes the unreadable link text on Woo, but the links are now slightly larger than the text. It's strange that it's not full-width on the screens that use the card styling (like Gravatar).

| Service | Before | After |
|---|---|---|
| Akismet | ![before-akismet](https://github.com/user-attachments/assets/523cadcc-c2aa-495c-8e02-c0d71bbaf143) | ![after-akismet](https://github.com/user-attachments/assets/c6072ac3-43f9-46ac-a42f-da4a713067a4) |
| Blaze Pro | ![before-blazepro](https://github.com/user-attachments/assets/ff139e77-7f99-4084-9e62-362827ddc7ad) | ![after-blazepro](https://github.com/user-attachments/assets/559bdd12-d393-403f-8ba4-88297e7b1143) |
| Crowdsignal | ![before-crowdsignal](https://github.com/user-attachments/assets/21d5bc40-6076-4025-b966-602a43abf95b) | ![after-crowdsignal](https://github.com/user-attachments/assets/109a644b-d17a-44e7-8910-002740087eb8) |
| Gravatar | ![before-gravatar](https://github.com/user-attachments/assets/cfa61d51-2f6c-427f-af92-0e571f1e58b9) | ![after-gravatar](https://github.com/user-attachments/assets/15baeaa8-5ad6-430e-bcb9-55b38fe137fe) |
| Jetpack Cloud | ![before-jetpackcloud](https://github.com/user-attachments/assets/2a3d5fc1-12ce-4126-987b-9d8857587aa6) | ![after-jetpackcloud](https://github.com/user-attachments/assets/e5205365-aa54-45fb-aa4c-1d263a47ae1b) |
| P2 | ![before-p2](https://github.com/user-attachments/assets/10848238-e8f1-4e50-be2b-c68d36c6d472) | ![after-p2](https://github.com/user-attachments/assets/a7eaa4c1-4255-4c20-9ee4-63c2b5182988) |
| VIP | ![before-vip](https://github.com/user-attachments/assets/5f112d48-9a47-468e-967a-a1c71c0f4f07) | ![after-vip](https://github.com/user-attachments/assets/e397f59f-11f5-43f3-8b63-675422f9f751) |
| Woo | ![before-woo](https://github.com/user-attachments/assets/fe629046-5a86-4acc-9ba0-2c79a321d469) | ![after-woo](https://github.com/user-attachments/assets/2694352e-c3be-403c-b268-5ca172066cd1) |
| WordPress.com | ![before-wpcom](https://github.com/user-attachments/assets/5a491d67-dbe7-485d-bf4a-1a1f6b7b7f24) | ![after-wpcom](https://github.com/user-attachments/assets/64f8d554-f697-4293-80ce-753bad5856d5) |
| WP Job Manager | ![before-wpjob](https://github.com/user-attachments/assets/bfbf34f9-9e87-4d29-afe9-c511af1dfcd1) | ![after-wpjob](https://github.com/user-attachments/assets/0e3a4816-4a99-4521-9f6b-d08c90d25047) |

chriskmnds committed:
feedback fixes

chriskmnds commented:
> I've gone through the list of different login screens (from https://github.com/Automattic/wp-calypso/pull/100807, h/t @adamwoodnz for pulling those together), and there are a few issues. The new compact notice is missing the bottom margin, so it collides with the title on a few of them. It also blends in on Crowdsignal & VIP's screens. This fixes the unreadable link text on Woo, but the links are now slightly larger than the text. It's strange that it's not full-width on the screens that use the card styling (like Gravatar).

My goodness, that's quite a few variations. 😵 

Thanks for testing @ryelle. I appreciate the effort and help. I think I've addressed your comments, roughly perhaps, except for:

> It's strange that it's not full-width on the screens that use the card styling (like Gravatar).

We can do that too. It may lose some of its subtleness stretching like the original, but we can do that. WDYT? I left a commented-out [width: 100%](https://github.com/Automattic/wp-calypso/pull/101254/files#diff-0daca1277c8be96d7a22626975ecb4ed35b75b193f04053c85273792186752bdR20) that you can switch to see more/less what to expect.

I need to do more testing / see where else this renders. We can tailor the design more if you have any suggestions. The cases you mentioned:

- Crowdsignal

<img width="400" alt="Screenshot 2025-03-13 at 6 49 46 PM" src="https://github.com/user-attachments/assets/32ac66fc-59bf-48d3-ba6e-f0488f2ffbcb" />

- VIP

<img width="400" alt="Screenshot 2025-03-13 at 6 49 58 PM" src="https://github.com/user-attachments/assets/a2c9dd7e-7752-4f83-84dc-53a8847836b3" />

- Woo 

<img width="400" alt="Screenshot 2025-03-13 at 6 50 21 PM" src="https://github.com/user-attachments/assets/953a486e-d9fd-43c0-8ff2-59220db4ac4d" />

chriskmnds commented:
p.s. I'll be visiting your [other suggestion](https://github.com/Automattic/wp-calypso/issues/100426#issuecomment-2718731880) in the issue separately / in another PR. @ryelle

ryelle committed:
Reverse logic for oauth page styles

ryelle commented:
> My goodness, that's quite a few variations. 😵

Yeah… There's an issue to redesign & unify at least all the oauth screens (https://github.com/Automattic/dotcom-forge/issues/9263), but until then we've got this 😑 

> We can do that too. It may lose some of its subtleness stretching like the original, but we can do that. WDYT? I left a commented-out [width: 100%](https://github.com/Automattic/wp-calypso/pull/101254/files#diff-0daca1277c8be96d7a22626975ecb4ed35b75b193f04053c85273792186752bdR20) that you can switch to see more/less what to expect.

Hm, it's awkward at 100% on more screens than not, so probably not worth it. It's better now that it's not sticking to the card container, at least.

I did push a commit to this PR, basically inverting the logic for the oauth styles— this way any generic oauth screen will get the light background, while the screens that we know are custom styled (Jetpack, Woo, Blaze, A4A) are left with the default notice background. Hopefully you don't mind  🙂 

This way it applies to any developer's oauth app:

![Screen Shot 2025-03-13 at 15 40 19](https://github.com/user-attachments/assets/ec41e639-b9b5-4bc0-9a6d-16619fa51d4d)

chriskmnds commented:
> I did push a commit to this PR, basically inverting the logic for the oauth styles— this way any generic oauth screen will get the light background, while the screens that we know are custom styled (Jetpack, Woo, Blaze, A4A) are left with the default notice background. Hopefully you don't mind 🙂

That's actually lovely. Thank you. 

I feel utterly defeated looking at all that login code and variations. I'd love to get my head around, but I'm just glad you are here right now 😁

chriskmnds committed:
cleanup

chriskmnds commented:
The one thing I noticed is with Crowsignal the background changes when in mobile view. However, I think that will be remedied with https://github.com/Automattic/wp-calypso/pull/101444 (befer/after below):

| Crowdsignal Now | Crowdsignal After |
|--------|--------|
| <img width="502" alt="Screenshot 2025-03-17 at 7 18 36 PM" src="https://github.com/user-attachments/assets/7760edc3-1395-4515-b1e3-f5841cd84e6c" /> | <img width="502" alt="Screenshot 2025-03-17 at 7 18 06 PM" src="https://github.com/user-attachments/assets/9f1c129f-7646-4415-989d-6bf122cbe9d2" /> |
